polyply/src/polyply_parser.py

```diff
@@ -17,12 +17,27 @@
 import networkx as nx
 import vermouth
 import vermouth.gmx
+from vermouth.parser_utils import SectionLineParser
 from vermouth.gmx.itp_read import ITPDirector
 
 class PolyplyParser(ITPDirector):
     '''
     Parser for polyply input format.
     '''
+    def __init__(self, force_field):
+        super().__init__(force_field)
+        self.citations = set()
+
+    @SectionLineParser.section_parser('moleculetype', 'citation')
+    def _parse_citation(self, line, lineno=0):
+        cite_keys = line.split()
+        self.current_block.citations.update(cite_keys)
+
+    @SectionLineParser.section_parser('citations')
+    def _pase_ff_citations(self, line, lineno=0):
```

Comment:
Is it correct one is 'citation' and the other is 'citation*s*'?

Reply:
yes, funny thing is you cannot have a directive both under molecultype and in general. They need to have a different name otherwise some strange behavior happens where 'randomly' one of the two is selected. I didn't figure it out but in the end I settled for naming the sections differently, which is consistent with the implementation in martinize

Reply:
Good enough fix for now, but could you open a Vermouth issue? Because it should most definitely not be 'randomly' selected.
The intended behaviour is to get the most specific section preferentially.

Although there is a gotcha:
Let's say the following section parsers are registered:
- a
- b
- b/a
- c

With the following document (and result)
```
[ a ]  ; matches a
[ b ]  ; b
[ a ]  ; b/a
[ c ]  ; c
[ a ]  ; a (since c closed b)
```

But note that this is *intended* behaviour. I'm not 100% sure it's working as intended.